File: backend/app/api/major_prospects.py
# backend/app/api/major_prospects.py
"""专业前景 API — 面向大一大二学生，按专业聚合真实就业/升学/考公数据。

数据全部带溯源（国家统计局 / 人社局工资价位 / 院校公开信息），纯规则聚合零 LLM。
"""

from fastapi import APIRouter, Depends, HTTPException, Query
from pydantic import BaseModel
from sqlalchemy.orm import Session

from app.database import get_db
from app.services import major_prospect_service as svc

router = APIRouter(prefix="/api/major-prospects", tags=["专业前景"])


class MajorListItem(BaseModel):
    name: str
    category: str
    source: str
    has_grad_intel: bool


class IndustrySalary(BaseModel):
    industry: str
    year: int
    salary_non_private: float
    salary_private: float | None
    vs_national: float
    source: str


class PositionSalary(BaseModel):
    position: str
    salary_median: int
    salary_min: int
    salary_max: int
    cities: list[str]
    source: str
    year: int


class CompanyItem(BaseModel):
    name: str
    industry: str
    size: str
    headquarters: str | None


class GradPathItem(BaseModel):
    school_name: str
    school_tier: str
    major_name: str
    year: int
    admission_ratio: str
    score_line: int | None
    push_ratio: str
    background_discrimination: str
    first_choice_protection: str


class CivilServiceInfo(BaseModel):
    level: str
    label: str
    note: str


class ProspectResponse(BaseModel):
    major: str
    matched_major: str
    exact_match: bool
    category: str
    industries: list[IndustrySalary]
    positions: list[PositionSalary]
    companies: list[CompanyItem]
    grad_paths: list[GradPathItem]
    civil_service: CivilServiceInfo
    related_majors: list[str]
    data_notes: list[str]


@router.get("/majors", response_model=list[MajorListItem])
def majors(db: Session = Depends(get_db)):
    """所有可选专业（映射表主干 + 已收录考研情报的专业）。"""
    return svc.list_majors(db)


@router.get("/detail", response_model=ProspectResponse)
def detail(
    major: str = Query(..., min_length=1, max_length=100, description="专业名称"),
    db: Session = Depends(get_db),
):
    """按专业聚合前景数据：行业薪资 / 岗位薪资 / 去向公司 / 考研路径 / 考公友好度。"""
    if not major.strip():
        raise HTTPException(status_code=422, detail="专业名称不能为空")
    return svc.get_prospect(db, major.strip())

File: frontend/lib/api/major-prospects.ts
import { request, buildQuery } from "./client";

/** 专业前景 — 按专业聚合真实就业/升学/考公数据 */

export interface MajorListItem {
  name: string;
  category: string;
  source: "mapped" | "grad_intel";
  has_grad_intel: boolean;
}

export interface IndustrySalary {
  industry: string;
  year: number;
  salary_non_private: number;
  salary_private: number | null;
  vs_national: number;
  source: string;
}

export interface PositionSalary {
  position: string;
  salary_median: number;
  salary_min: number;
  salary_max: number;
  cities: string[];
  source: string;
  year: number;
}

export interface ProspectCompany {
  name: string;
  industry: string;
  size: string;
  headquarters: string | null;
}

export interface GradPath {
  school_name: string;
  school_tier: string;
  major_name: string;
  year: number;
  admission_ratio: string;
  score_line: number | null;
  push_ratio: string;
  background_discrimination: string;
  first_choice_protection: string;
  outgoing_note: string | null;
  outgoing_risk: "friendly" | "acceptable" | "careful" | "high" | null;
}

export interface CivilServiceInfo {
  level: "high" | "medium" | "low";
  label: string;
  note: string;
}

export interface MajorProspect {
  major: string;
  matched_major: string;
  exact_match: boolean;
  category: string;
  industries: IndustrySalary[];
  positions: PositionSalary[];
  companies: ProspectCompany[];
  grad_paths: GradPath[];
  grad_personalized: boolean;
  civil_service: CivilServiceInfo;
  related_majors: string[];
  tier_fact: string;
  data_notes: string[];
}

/** 本科出身层次（用于升学路径个性化）。专科手动选档，不建专科院校库。 */
export const OUTGOING_TIERS = ["985", "211", "双一流", "一本", "二本", "专科"] as const;
export type OutgoingTier = (typeof OUTGOING_TIERS)[number];

// ===== 专业前景 =====
export const majorProspectApi = {
  majors: () => request<MajorListItem[]>("/api/major-prospects/majors"),
  outgoingTiers: () => request<{ tiers: string[] }>("/api/major-prospects/outgoing-tiers"),
  detail: (major: string, outgoingTier?: OutgoingTier | null) =>
    request<MajorProspect>(
      `/api/major-prospects/detail${buildQuery({
        major,
        ...(outgoingTier ? { outgoing_tier: outgoingTier } : {}),
      })}`,
    ),
};
